src/DocumentManager.h

```diff
@@ -71,6 +73,15 @@ class DocumentManager
 
     /** @brief Initialize function to create docMap_. */
     bool createDocumentMap();
+
+    /** @brief Update document in docMap_.If not exists,create one. */
+    void updateDocumentByID(int id);
+
+    /** @brief Retrive all histories from single document.
+     * @param message document id.
+     * @param message history list reference.
+    */
+    bool getHistoryFromDocument(int id,std::vector<Revision>& history);
```

Comment:
Input (id) should be const.
Output (history) should be a pointer.

Consider returning the output pointer instead. Return nullptr to denote failure.